content/en/serverless/google_cloud_run/_index.md

```diff
@@ -2,201 +2,298 @@
+
+Google Cloud Run is a fully managed serverless platform for deploying and scaling container-based applications. Datadog
+provides monitoring and log collection for Cloud Run through the [GCP integration][1]. Datadog also provides a solution,
+now in public beta, for instrumenting your Cloud Run applications with a purpose-built Agent to enable tracing, custom
+metrics, and direct log collection.
 
   <div class="alert alert-warning">This feature is in public beta. You can provide feedback through a <a href="https://forms.gle/HSiDGnTPvDvbzDAQA">feedback form</a>, or through your standard support channels. During the beta period, Cloud Run monitoring and APM tracing are available without a direct cost. Existing APM customers may incur increased span ingestion and volume costs. </div>
 
-## Tracing and custom metrics
-### Build your container
+## Getting started
 
-If you are using a Dockerfile to build your application, complete the following:
+### Prerequisites
 
-1. Instrument your application with a [supported Datadog tracing library][2]
+Make sure you have a [Datadog API Key][10] and are using a programming language [supported by a Datadog tracing library][2].
 
-2. Use the `COPY` instruction to copy the [Datadog `serverless-init` binary][3] into your Docker image.
+### Deploy a sample application in one click
 
-3. Use the `ENTRYPOINT` instruction to run the `serverless-init` binary as your Docker container is initiated.
+To deploy a sample application without the need to follow the rest of the guide, you can use one of [these examples][12]. The installation process asks required details, such as your Datadog API key. Note that these applications send data to the default website `datadoghq.com`.
 
-4. Use the `CMD` instruction to run your existing application and other required commands as arguments.
+### 1. Instrument your application
 
-The following are examples of how to complete these three steps. You may need to adjust these examples depending on your existing Dockerfile setup. 
+To build your container with Datadog instrumentation you can follow one of these two methods, depending if you are using a Dockerfile or buildpack.
 
+#### 1.a Instrument using Dockerfile
+
+You can instrument your application by adding the following lines to your Dockerfile. You may need to adjust these examples depending on your existing Dockerfile setup.
 
 {{< programming-lang-wrapper langs="go,python,nodejs,java,dotnet,ruby" >}}
 {{< programming-lang lang="go" >}}
+
 ```
-COPY --from=datadog/serverless-init:beta4 /datadog-init /app/datadog-init
+# copy the Datadog `serverless-init` into your Docker image
+COPY --from=datadog/serverless-init /datadog-init /app/datadog-init
+
+# change the entrypoint to wrap your application into the Datadog serverless-init process
 ENTRYPOINT ["/app/datadog-init"]
-CMD ["/path/to/your-go-binary"] (adapt this line to your needs)
+
+# optionally add Datadog tags
```

Comment:
do we want do add DD_SITE here?

Reply:
All the environment variables needed by Datadog can be defined either in the Dockerfile or in CloundRun Env settings, right?
Right now it's written below in the CloudRun config, since it has some more explanation and the reference to the website to find the value.
I think it's ok to leave it there, since it is a short guide and it works. Someone familiar with docker should understand that they can go either way.
Do you think it's worth to add a note that env variables can be placed in different places?